packages/checkout/widgets-lib/src/widgets/sale/hooks/useCurrency.ts

```diff
@@ -0,0 +1,170 @@
+    erc20Address: c.erc20_address,
+    exchangeId: c.exchange_id,
+  })),
+  currencyConversion: Object.entries(response.currency_conversion).reduce(
+    (acc, [key, value]) => {
+      acc[key] = {
+        amount: value.amount,
+        name: value.name,
+        type: value.type as SignPaymentTypes,
+      };
+      return acc;
+    },
+    {} as ClientConfigCurrencyConversion,
+  ),
+});
+
+type UseClientConfigParams = {
+  environment: Environment;
+  environmentId: string;
+  passport: Passport | undefined;
+  defaultCurrency?: string;
+};
+
+export const defaultClientConfig: ClientConfig = {
+  contractId: '',
+  currencies: [],
+  currencyConversion: {},
+};
+
+export const useCurrency = ({
+  environment,
+  environmentId,
+  passport,
+  defaultCurrency,
+}: UseClientConfigParams) => {
+  const [selectedCurrency, setSelectedCurrency] = useState<
+  ClientConfigCurrency | undefined
+  >();
+  const [clientConfig, setClientConfig] = useState<ClientConfig>(defaultClientConfig);
+  const [allCurrencies, setAllCurrencies] = useState<SaleWidgetCurrency[]>([]);
+
+  const checkoutClient = useCheckoutClient({ passport, environment });
+
+  useEffect(() => {
+    if (!environment || !environmentId) return;
+
+    const fetchSwappableCurrencies = async () => {
+      if (!checkoutClient || !passport) {
+        return [];
+      }
+      try {
+        const passportProvider = passport.connectEvm();
+        const zkEvmProvider = new providers.Web3Provider(passportProvider);
+        const checkoutNetworkInfo = await checkoutClient.getNetworkInfo({
+          provider: zkEvmProvider,
+        });
+        const swapAllowList = await checkoutClient.getTokenAllowList({
+          type: TokenFilterTypes.SWAP,
+          chainId: checkoutNetworkInfo.chainId,
+        });
+
+        return swapAllowList.tokens.map((token) => ({
+          ...token,
+          currencyType: SaleWidgetCurrencyType.SWAPPABLE,
+        }));
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching swappable currencies:', error);
+        return [];
+      }
+    };
+
+    const fetchSettlementCurrencies = async () => {
+      try {
+        const baseUrl = `${PRIMARY_SALES_API_BASE_URL[environment]}/${environmentId}/client-config`;
+        const response = await fetch(baseUrl, {
+          method: 'GET',
+          headers: { 'Content-Type': 'application/json' },
+        });
+
+        if (!response.ok) throw new Error(`${response.status} - ${response.statusText}`);
+
+        const data: ClientConfigResponse = await response.json();
+        const config = toClientConfig(data);
+        setClientConfig(config);
+
+        return config.currencies.map((currency) => ({
+          ...currency,
+          currencyType: SaleWidgetCurrencyType.SETTLEMENT,
+        }));
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching settlement currencies:', error);
```

Comment:
this may cause the widget to break.
Can you check if throwing here is captured by an error screen?

Reply:
This is how the `useClientConfig` is currently handling error - a `console.warn` and passing the empty default `clientConfig`. I'll add error handling for it to go straight to the error screen if the error occurs here as transak/coins flow both wouldn't work without `contractId`